packages/apollo/src/fetch-schema.ts

```diff
@@ -81,30 +110,35 @@ export const fetchSchema = async (
     options.fetchOptions = { agent: agent };
   }
 
-  return toPromise(
-    // XXX node-fetch isn't compatible typescript wise here?
-    execute(createHttpLink(options), {
+  const { data, errors }: any = await toPromise(
+    linkExecute(createHttpLink(options), {
       query: introspection,
       context: { headers }
     })
-  ).then(({ data, errors }: any) => {
-    if (errors)
-      throw new Error(errors.map(({ message }: Error) => message).join("\n"));
-    return buildClientSchema({ __schema: data.__schema });
-  });
+  );
+
+  if (!data) {
```

Comment:
Should the `data` and `error` checks be reversed, so we always report errors if there are any?

Reply:
Makes sense to me.  Fixed via 9be573a45302af8302cffb6de36f61e6be721b41.